debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/valueholder/LogMinerDmlEntryImpl.java

```diff
@@ -28,12 +31,29 @@
     private String rowId;
     private int sequence;
 
-    public LogMinerDmlEntryImpl(int operation, List<LogMinerColumnValue> newLmColumnValues, List<LogMinerColumnValue> oldLmColumnValues) {
+    private LogMinerDmlEntryImpl(int operation, List<LogMinerColumnValue> newLmColumnValues, List<LogMinerColumnValue> oldLmColumnValues) {
         this.operation = operation;
         this.newLmColumnValues = newLmColumnValues;
         this.oldLmColumnValues = oldLmColumnValues;
     }
 
+    public static LogMinerDmlEntry forInsert(List<LogMinerColumnValue> newLmColumnValues) {
+        return new LogMinerDmlEntryImpl(RowMapper.INSERT, newLmColumnValues, Collections.emptyList());
+    }
+
+    public static LogMinerDmlEntry forUpdate(List<LogMinerColumnValue> newLmColumnValues, List<LogMinerColumnValue> oldLmColumnValues) {
+        return new LogMinerDmlEntryImpl(RowMapper.UPDATE, newLmColumnValues, oldLmColumnValues);
+    }
+
+    public static LogMinerDmlEntry forDelete(List<LogMinerColumnValue> oldLmColumnValues) {
+        return new LogMinerDmlEntryImpl(RowMapper.DELETE, Collections.emptyList(), oldLmColumnValues);
+    }
+
+    public static LogMinerDmlEntry forLobLocator(List<LogMinerColumnValue> newLmColumnValues) {
+        // TODO: can that copy be avoided?
```

Comment:
Yes.  

There is a use case where a user may update two or more BLOB/CLOB fields only where one of those fields are updated with a small enough value that LogMiner treats it inlined and the other where it must be chunked.  The rows mined would look something like the following:

```
SCN=1000 SEL_LOB_LOCATOR - update set "LARGE_CLOB"=lob_c where "ID"='1';
<lob chunk rows, i.e. LOB_WRITE/LOB_TRIM>
SCN=1002 UPDATE - update set "SMALL_CLOB"='SmallValue' where "ID"='1';
```
The transaction reconciliation step takes the entries in these arrays and merges other events values into the first event and discards those of the latter events where merging happens.  Here the `UPDATE` will get merged into the previous `SEL_LOB_LOCATOR` event.  If we didn't have the copy, when we added the "SMALL_CLOB" entry to the new values of the `SEL_LOB_LOCATOR`, then it would also be present in the old values, which isn't correct.

Reply:
Hum, hum. I see.